kolibri/plugins/coach/class_summary_api.py

```diff
@@ -80,8 +81,10 @@ def get_status(log):
                     return HELP_NEEDED
         if log["progress"] == 1:
             return COMPLETED
-        elif log["progress"] == 0:
-            return NOT_STARTED
+        if log["kind"] == content_kinds.EXERCISE:
+            # if there are no attempt logs for this exercise, status is NOT_STARTED
+            if logger_models.AttemptLog.objects.filter(user_id=log['user_id'], sessionlog__content_id=log['content_id']).count() == 0:
```

Comment:
uh oh - this adds a bunch of extra DB calls to every `class_summary_api` request... I'm a little worried about impact on performance

Reply:
Just a hunch, but I wonder if it would be faster to do this as an annotation on the original `content_log` query rather than doing it individually per-log here? Would cool to see some time measurements